core/src/main/scala/org/broadinstitute/dsde/rawls/bucketMigration/BucketMigrationService.scala

```diff
@@ -27,33 +30,173 @@ class BucketMigrationService(val dataSource: SlickDataSource, val samDAO: SamDAO
     with WorkspaceSupport
     with LazyLogging {
 
-  def getBucketMigrationProgressForWorkspace(
-    workspaceName: WorkspaceName
-  ): Future[Option[MultiregionalBucketMigrationProgress]] =
+  /**
+    * Helper functions to enforce appropriate authz and load workspace(s) if authz passes
+    */
+  def asFCAdminWithWorkspace[T](workspaceName: WorkspaceName)(op: Workspace => Future[T]): Future[T] =
```

Comment:
These and the `Internal` functions below had to be made public to make unit testing easier and less repetitive. This is one of the things I'm questioning

Reply:
How is this "asFCAdmin"?

Reply:
🤔 Getting my head around this. I see that the tests are call these methods to ensure that we are properly disallowing access in the correct situations which makes sense. I further observe that the consuming methods within `BucketMigrationService` like `migrateWorkspaceBucketsInBillingProject(billingProjectName)` do not have corresponding tests. Would it make sense to get a test around those instead and hide these methods?

That being said, I think this is an "ok" implementation as-is. The API is sound and nothing is preventing us from refactoring this in the future.

Reply:
Yeah, I made them public because I was trying to avoid a lot of repetitive tests, but I'm thinking now that a bit of repetition in the tests would be preferred if we can actually directly test the real entrypoints to this service